### PR TITLE
#3284 - Create a new message for an unconfirmed account with a link

### PR DESCRIPTION
**Pull request by IrinaZhadzinets**
**Related issue(s):**
* Fixes https://github.com/scandipwa/scandipwa/issues/3284

**In this PR:**
* Changed the message if the account is not confirmed

### CODE REVIEW

File: src/Model/CustomerTokenService.php [hunk 1]
```diff
@@ -67,6 +68,11 @@ class CustomerTokenService extends SourceCustomerTokenService
      */
     protected $requestThrottler;
 
+    /**
+     * @var UrlInterface
+     */
+    protected $urlBuilder;
```
Comment (aleksandrsm) [inline src/Model/CustomerTokenService.php:74]: if for latest PWA, can have a strict class type 
`protected UrlInterface $urlBuilder;`

File: src/Model/CustomerTokenService.php [hunk 2]
```diff
@@ -106,16 +115,21 @@ public function __construct(
      * @param $password
      * @return string
```
Comment (aleksandrsm) [inline src/Model/CustomerTokenService.php:116]: update docs of the method

File: src/Model/CustomerTokenService.php [hunk 3]
```diff
@@ -106,16 +115,21 @@ public function __construct(
...
     {
         $this->validatorHelper->validate($username, $password);
         $this->getRequestThrottler()->throttle($username, RequestThrottler::USER_TYPE_CUSTOMER);
 
         try {
             $customerDataObject = $this->accountManagement->authenticate($username, $password);
         } catch (EmailNotConfirmedException $e) {
+            $href = $this->urlBuilder->getUrl(
+                'customer/account/confirmation',
+                ['_query' => ['store' => $storeId, 'email' => $username]]
```
Comment (aleksandrsm) [inline src/Model/CustomerTokenService.php:128]: not 100% certain, but you may want to check if $storeId is not null here

**Review by aleksandrsm — CHANGES_REQUESTED** [review]
see comments

File: src/Model/CustomerTokenService.php (hunk 3 above)
Comment (IrinaZhadzinets) [reply src/Model/CustomerTokenService.php:128]: Do I need to add separate requests (with and without store) if it returns the correct URL even if storeId=null(in this case, the address for the default store is returned)?

Comment (aleksandrsm) [reply src/Model/CustomerTokenService.php:129]: if it works fine with null, then never-mind, can keep as is

**Review by aleksandrsm — APPROVED** [review]

**Pull request merged**